ios/sdk/src/JitsiMeetView.m

```diff
@@ -87,157 +66,116 @@ - (instancetype)initWithFrame:(CGRect)frame {
-    [externalAPI sendSetVideoMuted:muted];
+    [self.jitsiMeetViewController setVideoMuted:muted];
 }
 
 - (void)setClosedCaptionsEnabled:(BOOL)enabled {
     ExternalAPI *externalAPI = [[JitsiMeet sharedInstance] getExternalAPI];
     [externalAPI sendSetClosedCaptionsEnabled:enabled];
 }
 
-#pragma mark Private methods
-
-/**
- * Passes the given props to the React Native application. The props which we pass
- * are a combination of 3 different sources:
- *
- * - JitsiMeet.defaultConferenceOptions
- * - This function's parameters
- * - Some extras which are added by this function
- */
-- (void)setProps:(NSDictionary *_Nonnull)newProps {
-    NSMutableDictionary *props = mergeProps([[JitsiMeet sharedInstance] getDefaultProps], newProps);
-
-    // Set the PiP flag if it wasn't manually set.
-    NSMutableDictionary *featureFlags = props[@"flags"];
-    if (featureFlags[PiPEnabledFeatureFlag] == nil) {
-        featureFlags[PiPEnabledFeatureFlag]
-            = [NSNumber numberWithBool:
-               self.delegate && [self.delegate respondsToSelector:@selector(enterPictureInPicture:)]];
-    }
+#pragma mark Private
 
-    props[@"externalAPIScope"] = externalAPIScope;
-
-    // This method is supposed to be imperative i.e. a second
-    // invocation with one and the same URL is expected to join the respective
-    // conference again if the first invocation was followed by leaving the
-    // conference. However, React and, respectively,
-    // appProperties/initialProperties are declarative expressions i.e. one and
-    // the same URL will not trigger an automatic re-render in the JavaScript
-    // source code. The workaround implemented below introduces imperativeness
-    // in React Component props by defining a unique value per invocation.
-    props[@"timestamp"] = @(mach_absolute_time());
-
-    if (rootView) {
-        // Update props with the new URL.
-        rootView.appProperties = props;
-    } else {
-        RCTBridge *bridge = [[JitsiMeet sharedInstance] getReactBridge];
-        rootView
-            = [[RNRootView alloc] initWithBridge:bridge
-                                      moduleName:@"App"
-                               initialProperties:props];
-        rootView.backgroundColor = self.backgroundColor;
-
-        // Add rootView as a subview which completely covers this one.
-        [rootView setFrame:[self bounds]];
-        rootView.autoresizingMask
-            = UIViewAutoresizingFlexibleWidth
-                | UIViewAutoresizingFlexibleHeight;
-        [self addSubview:rootView];
-    }
+- (BOOL)isPiPEnabled {
+    return self.delegate && [self.delegate respondsToSelector:@selector(enterPictureInPicture:)];
 }
 
-+ (BOOL)setPropsInViews:(NSDictionary *_Nonnull)newProps {
-    BOOL handled = NO;
+- (void)registerObservers {
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleSendEventNotification:) name:sendEventNotificationName object:nil];
```

Comment:
@saghul 

I think this function has bug.
Our Swift iOS app using SDK like this
```
class JitsiViewController: UIViewController {
    ....
    jitsiMeetView = JitsiMeetView()
    jitsiMeetView?.delegate = self
    ....
}
extension JitsiViewController: JitsiMeetViewDelegate {
    func conferenceWillJoin(_ data: [AnyHashable : Any]!) {
    }
}
```

Problem is `jitsiMeetView = JitsiMeetView()` called both JitsiMeetView.init and JitsiMeetView.initWithFrame (one from [super init])
-> called initWithXXX 2 times
-> called registerObservers 2 times
-> It will make `conferenceWillJoin` called 2 times duplicate when fire 1 event CONFERENCE_WILL_JOIN

I have fixed by add
```
- (void)registerObservers {
    #remove before add
    [[NSNotificationCenter defaultCenter] removeObserver:self];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleUpdateViewPropsNotification:) name:updateViewPropsNotificationName object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(handleSendEventNotification:) name:sendEventNotificationName object:nil];
 }
```

It work but not sure if will break something else, I not good at iOS yet, could you please confirm?

Reply:
Oh, is that so? We should make sure initWithXXX (terrible function name, I know) is only called once.

Mind sending a PR?